### PR TITLE
3.10 adjustments for clang-16

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

*3.10 adjustments for clang-16*

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: 3rdParty/s2geometry/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libraries" FORCE)
 set(BUILD_EXAMPLES OFF CACHE BOOL "Build examples" FORCE)
 set(INSTALL_HEADERS OFF CACHE BOOL "Install headers" FORCE)
 
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/${ARANGO_S2GEOMETRY_VERSION} EXCLUDE_FROM_ALL)
+add_subdirectory(${ARANGO_S2GEOMETRY_VERSION})
```
Comment (MBkkt) [inline 3rdParty/s2geometry/CMakeLists.txt:20]: It's not correct without this https://github.com/arangodb/arangodb/blob/devel/CMakeLists.txt#L1190

**Review by MBkkt — COMMENTED** [review]
Also making abseil submodule require these lines before add_subdirectory(...)

```
UpdateModule(${GIT_EXECUTABLE} "abseil-cpp" ${CMAKE_CURRENT_SOURCE_DIR} "README.md")

set(ABSL_PROPAGATE_CXX_STD ON)
```

**Review by KVS85 — CHANGES_REQUESTED** [review]
clang isn't used as a production compiler for 3.10.

**Review by MBkkt — CHANGES_REQUESTED** [review]
Check my previous comments

**Pull request closed without merging**